client/landing/jetpack-cloud/components/scan-threats/index.tsx

```diff
@@ -44,27 +45,52 @@ const ScanThreats = ( { site, threats }: Props ) => {
 	const dispatch = useDispatch();
 
 	const openFixAllThreatsDialog = React.useCallback( () => {
+		dispatch(
+			recordTracksEvent( `calypso_scan_all_threats_dialog_open`, {
+				site_id: site.ID,
+			} )
+		);
 		setShowFixAllThreatsDialog( true );
-	}, [] );
+	}, [ dispatch, site ] );
 
-	const openDialog = React.useCallback( ( action: ThreatAction, threat: Threat ) => {
-		setSelectedThreat( threat );
-		setActionToPerform( action );
-		setShowThreatDialog( true );
-	}, [] );
+	const openDialog = React.useCallback(
+		( action: ThreatAction, threat: Threat ) => {
+			dispatch(
+				recordTracksEvent( `calypso_scan_${ action }_threat_dialog_open`, {
+					site_id: site.ID,
+					threat_id: threat.id,
+				} )
+			);
+			setSelectedThreat( threat );
+			setActionToPerform( action );
+			setShowThreatDialog( true );
+		},
+		[ dispatch, site ]
+	);
 
 	const closeDialog = React.useCallback( () => {
 		setShowThreatDialog( false );
 	}, [] );
 
 	const confirmAction = React.useCallback( () => {
+		dispatch(
+			recordTracksEvent( `calypso_scan_threat_${ actionToPerform }`, {
+				site_id: site.ID,
+				threat_id: selectedThreat.id,
```

Comment:
I don't think we need to record threat_id here. But is there anyting else in the thread that could be useful as a grouping here? Is there a threat slug or some sort of threat type that we can record?

Reply:
Not sure right know but I'll look into it. Is this because, for our analytics purposes, we don't really care about unique instances, and instead, we care about the aggregations?

Reply:
The only thing I saw we could use is the threat signature but I don't know how unique that value is. I think it could help us find common threat causes.

In any case, I think we should revisit this later on when we have more clarity of the end shape/data of a threat.